sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/keyvault_deleted_secret.hpp

```diff
@@ -9,14 +9,20 @@
 #pragma once
 #include <azure/core/datetime.hpp>
 #include <azure/keyvault/secrets/keyvault_secret.hpp>
+namespace Azure { namespace Security { namespace KeyVault { namespace Secrets { namespace _detail {
+  struct DeletedSecretSerializer;
+  struct DeletedSecretPagedResultSerializer;
+}}}}} // namespace Azure::Security::KeyVault::Secrets::_detail
 
 namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
   /**
    * @brief A Deleted Secret consisting of its previous id, attributes and its tags,
    * as well as information on when it will be purged.
    */
-  struct DeletedSecret : public Secret
+  struct DeletedSecret : public KeyVaultSecret
   {
+    friend struct Azure::Security::KeyVault::Secrets::_detail::DeletedSecretSerializer;
+    friend struct Azure::Security::KeyVault::Secrets::_detail::DeletedSecretPagedResultSerializer;
```

Comment:
I would just let ScheduledPurgeDate and DeletedOn as public members to avoid these `friends`

or I would add an extra public constructor which takes the two values.  Generally, we try to minimize as much as possible using friend access.

Reply:
good point , i'll take them out